patches/minecraft/net/minecraft/village/Village.java.patch

```diff
@@ -45,6 +45,15 @@
  
                  if (entityplayer1 != null)
                  {
+@@ -362,7 +364,7 @@
+                 villagedoorinfo.func_75466_d();
+             }
+ 
+-            if (!this.func_179860_f(villagedoorinfo.func_179852_d()) || Math.abs(this.field_75581_g - villagedoorinfo.func_75473_b()) > 1200)
++            if (!field_75586_a.func_175667_e(villagedoorinfo.func_179852_d()) || !this.func_179860_f(villagedoorinfo.func_179852_d()) || Math.abs(this.field_75581_g - villagedoorinfo.func_75473_b()) > 1200) // Forge: check that the door block is loaded to avoid loading chunks
```

Comment:
I can't test it right now, but wouldn't this lead to doors being removed from villages when their chunk unloads (and possibly villages being permanently destroyed/recreated when they completely unload)? Or is there some code that re-adds doors as soon as the chunks are loaded? My fix yesterday was adding the check to isWoodenDoor and returning true for unloaded chunks.

Reply:
Yeah, you're right, I got this logic a bit backwards. If the chunk is not loaded, it should just assume the door is fine.